src/pip/_internal/commands/list.py

```diff
@@ -213,14 +213,20 @@ def latest_info(dist):
             # This is done for 2x speed up of requests to pypi.org
             # so that "real time" of this function
             # is almost equal to "user time"
-            pool = Pool(DEFAULT_POOLSIZE)
-
-            for dist in pool.imap_unordered(latest_info, packages):
-                if dist is not None:
-                    yield dist
-
-            pool.close()
-            pool.join()
+            try:
+                pool = Pool(DEFAULT_POOLSIZE)
+
+                for dist in pool.imap_unordered(latest_info, packages):
+                    if dist is not None:
+                        yield dist
+
+                pool.close()
+                pool.join()
+            except ImportError:
```

Comment:
I am not an expert of when Python actually imports class `Pool` and raises `ImportError` but in my opinion it should be in the actual importing code:
```
from multiprocessing.dummy import Pool
```

In CPython they wrap importing of modue in try except clause.
(https://github.com/python/cpython/blob/d9a43e20facdf4ad10186f820601c6580e1baa80/Lib/multiprocessing/synchronize.py#L24-L33)

So the except part will never be executed.

Reply:
Also it might be cleaner to move both for loops in a `fetch_latest_info_multithreaded` and `fetch_latest_info_single_threaded`, and then include the `from multiprocessing.dummy import Pool` in the `fetch_latest_info_multithreaded` function

Reply:
Yeah, I think the ImportError block needs to be put on the `from multiprocessing.dummy` line instead.

Reply:
yeah I was wondering if this reliance on delayed import would be allowed, I'll change it if it's really necessary
I'm not sure what you mean by "the except part will never be executed" though @CrafterKolyan, since that exception is exactly what caused his issue,. raising from within an except block still raises.

Reply:
I'm not exactly sure from the original bug report which statement is triggering the `ImportError`. We should probably determine that, and just wrap that in the `try...except` (maybe setting a flag that we test later to choose the right code path).

Also, should this code have a test? Otherwise there's a (small) risk that the two implementations get out of line.